### ISSUE TITLE
Region dropdown not displaying custom options on first load when country is pre-set

### ISSUE THREAD

**Issue by theevann** [country-regions/react-country-region-selector#79]
On the first load of the component, if country is pre-set, then the region component is not displaying custom options.

To re-use the example :
```jsx
import React, { Component } from 'react';

// note that you can also export the source data via CountryRegionData. It's in a deliberately concise format to 
// keep file size down
import { CountryDropdown, RegionDropdown, CountryRegionData } from 'react-country-region-selector';


class Example extends Component {
  constructor (props) {
    super(props);
    this.state = { country: 'United States', region: '' };
  }

  selectCountry (val) {
    this.setState({ country: val });
  }

  selectRegion (val) {
    this.setState({ region: val });
  }

  render () {
    const { country, region } = this.state;
    return (
      <div>
        <CountryDropdown
          value={country}
          onChange={(val) => this.selectCountry(val)} />
        <RegionDropdown
          country={country}
          value={region}
          onChange={(val) => this.selectRegion(val)}
          customOptions={['-- Unknown']} />
      </div>
    );
  }
}
```

**Comment by Mnabawy** [comment]
hey @theevann if you are using hooks why you don't use function component

**Comment by ernestpascual** [comment]
I encountered the same thing so I tried firing a button that changes the state of the country for the region dropdown and the customOptions showed. 

I also peeked the [code](https://github.com/country-regions/react-country-region-selector/blob/527e840c8e12657e1aa92feb1ae68da2c5b2d89b/src/RegionDropdown.js#L15) and saw `...this.getCustomOptions(defaultRegions)` in  componentDidUpdate 😂 

To solve this without doing fixing the code, I guess you can add a little delay and then change the state (or something that would trigger the component to re-render).

**Comment by benkeen** [comment]
Oof, sorry guys. This was resolved in `4.0.0`.
